packages/capabilities/src/access.js

```diff
@@ -108,3 +108,90 @@ export const claim = base.derive({
   }),
   derives: equalWith,
 })
+
+// https://github.com/web3-storage/specs/blob/main/w3-access.md#accessdelegate
+export const delegate = base.derive({
+  to: capability({
+    can: 'access/delegate',
+    /**
+     * Field MUST be a space DID with a storage provider. Delegation will be stored just like any other DAG stored using store/add capability.
+     *
+     * @see https://github.com/web3-storage/specs/blob/main/w3-access.md#delegate-with
+     */
+    with: DID.match({ method: 'key' }),
+    nb: {
+      // keys SHOULD be CIDs, but we won't require it in the schema
+      /**
+       * @type {Schema.Schema<AccessDelegateDelegations>}
+       */
+      delegations: Schema.dictionary({
+        value: Schema.Link.match(),
+      }),
+    },
+    derives: (claim, proof) => {
+      return (
+        fail(equalWith(claim, proof)) ||
+        fail(subsetsNbDelegations(claim, proof)) ||
+        true
+      )
+    },
+  }),
+  derives: (claim, proof) => {
+    // no need to check claim.nb.delegations is subset of proof
+    // because the proofs types here never include constraints on the nb.delegations set
+    return fail(equalWith(claim, proof)) || true
+  },
+})
+
+/**
+ * @typedef {Schema.Dictionary<string, Types.Link<unknown, number, number, 0 | 1>>} AccessDelegateDelegations
+ */
+
+/**
+ * Parsed Capability for access/delegate
+ *
+ * @typedef {object} ParsedAccessDelegate
+ * @property {string} can
+ * @property {object} nb
+ * @property {AccessDelegateDelegations} [nb.delegations]
+ */
+
+/**
+ * returns whether the claimed ucan is proves by the proof ucan.
+ * both are access/delegate, or at least have same semantics for `nb.delegations`, which is a set of delegations.
+ * checks that the claimed delegation set is equal to or less than the proven delegation set.
+ * usable with {import('@ucanto/interface').Derives}.
+ *
+ * @param {ParsedAccessDelegate} claim
+ * @param {ParsedAccessDelegate} proof
+ */
+function subsetsNbDelegations(claim, proof) {
+  /** @param {ParsedAccessDelegate} ucan */
+  const nbDelegationsCids = (ucan) =>
+    new Set(Object.values(ucan.nb.delegations || {}).map(String))
+  const missingProofs = setDifference(
+    new Set(nbDelegationsCids(claim)),
```

Comment:
You create a set and then pass it into a set constructor, which seems unnecessary

Reply:
https://github.com/web3-storage/w3protocol/pull/420/commits/d03e081ef3bfda4e2aeaa01e09aded00b27c5efa